### ISSUE TITLE
DRM Videos is not playing and Play DRM-controlled content is check

### ISSUE THREAD

**Issue by FlashAmarillo** [zen-browser/desktop#715]
### What happened?

The videos of the UDEMY platform cannot be played because they are protected content and despite having the option “Play DRM-controlled content” checked, it is still not possible to play the videos. Even in the extensions section, the “Widevine Content Decryption Module provided by Google Inc.” is working correctly and is always enabled. Is it possible that this feature is not working correctly?

![2024-08-25_21h25_25](https://github.com/user-attachments/assets/914b9099-217c-4c45-86db-4c2fe4ecb466)
![2024-08-25_21h25_45](https://github.com/user-attachments/assets/e24824d5-bdde-4b8b-b03b-7b783cd4096e)
![2024-08-25_21h26_22](https://github.com/user-attachments/assets/27678c22-a56f-4666-9456-2452d83e6d55)

Can you help me to solve this problem with the browser or is there any solution I can do apart from using another browser like firefox where it works perfectly and the videos can be played correctly?

### Reproducible?

- [x] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.0-a.29

### Severity impact

High

### What platform are you seeing the problem on?

Windows

**Comment by notsatvrn** [comment]
This has been [addressed in the FAQ](https://docs.zen-browser.app/faq#why-cant-zen-browser-play-drm-protected-content).
TL;DR: Until Zen obtains a pricey Widevine license, playback of DRM-protected content is prohibited. Use a different browser or help financially support development on Patreon or Ko-fi.

**Comment by tbodt** [comment]
I am curious, is this pricing information based on reaching out to Widevine for a quote? I couldn't find info on how much it would cost on their website.

**Comment by ethpsantos** [comment]
> This has been [addressed in the FAQ](https://docs.zen-browser.app/faq#why-cant-zen-browser-play-drm-protected-content). TL;DR: Until Zen obtains a pricey Widevine license, playback of DRM-protected content is prohibited. Use a different browser or help financially support development on Patreon or Ko-fi.

That is not the right way to do it imo.
You don't need to buy a Widevine license because it is free... and if you are trying to pay something is to a 3rd-party company to proxy the Google Widevine DRM for you... so you are paying for the proxy services not the Google Widevine DRM licenses.

So why forks developers have so much issues to enable DRM in their browsers?

Because you need an company behind your browser to sign the license agreement where that company is legally liable if the DRM got cracked in your browser... so any individual (person) account can't access the Google Widevine DRM services.
You need a Google account tied to a company.

So you have two paths here:

1) Correct patch: create a company in name of your browser that will be the owner of it... so you can have full access to Google Widevine DRM Cloud services and use it in the browsers maintained by that company.

2) The proxy way: pay a 3rd-party company to proxy the Google Widevine DRM Cloud requests for you... the company will ask a big amount of money because they are liable for any crash happening with your browser.

And to be fair to have a registered company is way cheaper than paying a 3rd-party to proxy it for you.

PS. I'm posting it here because I saw a lot of misunderstanding in the Reddit about Floorp and Zen. No. There is no license fee to use Google Widevine DRM... the FAQ is not exactly correct and don't explain that the $5000 anual price is to pay a 3rd-party proxy and not Widevine fee at all.

**Comment by bdbch** [comment]
Good explanation @ethpsantos – don't know why people would downvote you. This is the correct way to get access to the Widevine license. The 5k fee is a third party contractor taking the liability for you.

I'm looking forward for Zen to get DRM support eventually. Right now it won't be my main browser for a few reasons but it has a lot of potential for growth.

**Comment by YamiYukiSenpai** [comment]
Won't there be regulations involved in creating a company?  If it's as easy as that, wouldn't other forks have done this already?

**Comment by kernelb00t** [comment]
Hey, so how does Waterfox, Gnome Web and other Firefox based derivatives get a license? They are getting a license on their own, or are they using Mozilla's one? Because this makes no sense. Zen browser is not rewriting Gecko entirely, it is not that far from Firefox in the backend.

Also, what are the conditions to make a firefox clone? We can't use Mozilla's widevine module? It seems that one can build Firefox and get the module without restrictions, isn't it?

I understand we, users, will never get an official Widevine support until the license problem is resolved (~~fk you google~~). We would need to resolve this by using Zenivine, but this would violate every policy and mayybee make Mozilla lock the access to their Widevine module, right?

**Comment by teohhanhui** [comment]
https://www.reddit.com/r/zen_browser/comments/1hb869o/comment/m1e9zu3/

**Comment by mjbrm** [comment]
Maubg is simply trying to scam people into giving him $5000 when he could set up a registered company and get the Widevine license for free.  I've set up an LLC in the USA, it's really not that hard.

**Comment by 12th-devs** [comment]
@mjbrm Maibg has denied starting fundraisers multiple times as he feels it is "financially irresponsible" for him to buy the license

**Comment by mjbrm** [comment]
@Anoms12 he doesn't have to buy the license like @ethpsantos says. I've done it for my shitty little private fork. If I can do it he can too. There's no excuse.

**Comment by mjbrm** [comment]
@Anoms12 take maubg's nuts out of your mouth.

**Comment by 12th-devs** [comment]
@mjbrm how did you achieve this? You mention an LSC? I understand you would like DRM but this guy is only in college and prob doesn't know how to (which you would know if you had done a _bit_ more research on.) So maybe instead of fighting we can ping him with some documentation on how to do this or start a PR?

**Comment by mr-cheffy** [comment]
Please keep the comments civil, thanks. We've spoken to widevine and it's 9k license fee + we need to be a company with a set of employees anyways. That's my understanding at least.

And please DO NOT make people think im trying to scam anyone, we are talking to the linux foundation for a grant to pay this license, we are not asking for money / donations (in relation to widevine, it's unrelated).

**Comment by greeeen-dev** [comment]
> [@Anoms12](https://github.com/Anoms12) take maubg's nuts out of your mouth.

@mjbrm Perhaps contribute in a constructive way instead of attacking people. Your comment helps absolutely nobody.

**Comment by mr-cheffy** [comment]
> If I can do it he can too.

Oh, you've acquired a widevine license? Could you please send me some sort of documentation or build that can help us understand this process a bit more?

> We can't use Mozilla's widevine module?

Yes, we are using it indeed, but we need zen to be signed with a widevine license for that unfortunately

But im not really informed into this matter, so please don't quote me on this. Ill let Nicholas (@logicsec) handle this one as he is more informed on the topic. Thanks

**Comment by nocanoa** [comment]
@mjbrm This is not the place to talk like this, also doesn't seem like this is the first time u act like this [other issue](https://github.com/fedora-silverblue/issue-tracker/issues/640), and ur 2 day account doesn't invoke any trust ether.

Also not everyone lives in the USA.
If u don't like it just use ur "shitty little private fork", and don't annoy other people doing their best.

**Comment by mauro-balades** [comment]
> We can't use Mozilla's widevine module?

Yes, we are using it indeed, but we need zen to be signed with a widevine license for that unfortunately

**Comment by logicsec** [comment]
I wanted to comment here just to clear up any confusion on the different styles of DRM content sites like Netflix and Udemy utilize a variant of DRM that Zen currently does not support. This variant requires a server that is owned by Google to sign requests in order to unlock the content. This variant is not the same thing that you see in Zen currently that is also labeled as widevine. 

The current version that you see inside of Zen browser is to unlock level one or basic DRM content. In order for us to unlock level three DRM content we will need to pay about $9000 to get everything set up.

Also, please note that the current iteration of Zen browser is still in Beta and there will be features missing. Attacking anyone on the Dev team or any staff that support GitHub or discord is not appreciated and just makes their jobs harder. Please keep that in mind when posting in the future, thank you.

EDIT: Getting the license directly from Google has never and will never be an option. While widevine is technically free, their contract requires that you use a 3rd party to access the service. This 3rd party charges an implementation fee and a yearly upkeep license. The fact someone in here is stating that you just need a company and you get it free is false. My company is worth 500million dollars and we still needed to pay for the 3rd implementation partner. Anyone that has used Google in a business setting knows that Google does not work with companies directly.

**Comment by logicsec** [comment]
@kernelb00t I know for Waterfox, they got a sponsor for Widevine. Originally, they were using a Widevine patcher to get around paying for the implementation. Netflix, and I assume others, blocked this patching and added a layer of security on top of L3 DRM proxies to protect their content. You can see a lot of this discussion on the video downloaded forum. Just search Netflix.

**Comment by kernelb00t** [comment]
Ok, so the best option for Zen would be a grant/sponsorship by the Linux foundation, Mozilla or Google for that. 

@mauro-balades Concerning the Linux Foundation grant, what is the current state of it? Is it close to being approved, or it is just in talking stage?

**Comment by mjbrm** [comment]
Yall are dumb as shit. @ethpsantos spelled it out. If @mauro-balades can set up an entire ass github and fork a browser then you can figure out how to do it yourself without documentation. you don't have to pay 9k. It's 9k now instead of 5k?? @logicsec bullshit. you guys are playing some sketchy games here.

**Comment by mjbrm** [comment]
@nocanoa the process is the same regardless of country. my account age doesn't change the facts of what i've said. i don't use my shitty little fork, i just did it to see if i could because I got sick of @mauro-balades crying all over the internet about fees. I use good old Firefox not some browser made by a dude in his car

**Comment by Anthonyy232** [comment]
edit - N/A, user being addressed got banned so..
